### PR TITLE
[Jax] Drop redundant functions and use `manifest_utils` for helpers

### PR DESCRIPTION
**Pull request by erman-gurses**
## Motivation

Follow up PR after: https://github.com/ROCm/TheRock/pull/2866
Use shared filename normalization helpers to reduce duplication. 
Both `generate_jax_manifest.py` and `upload_jax_manifest.py` implemented similar logic for normalizing Python versions and git refs when constructing manifest filenames.

## Technical Details

- Use shared helpers from manifest_filename_utils:
  - normalize_python_version_for_filename
  - normalize_ref_for_filename 
- Remove duplicate normalization logic previously implemented locally in the scripts.
- Update unit tests to use the shared helpers.

## Test Plan
Test local and on CI

### Local
```
TheRock$  python -m unittest build_tools/github_actions/tests/upload_jax_manifest_test.py
Manifest expected at: /tmp/tmpsdxcx_dz/manifests/therock-manifest_jax_py3.12_nightly.json
Retrieving bucket info...
  (implicit) github_repository: ROCm/TheRock
  (implicit) is_pr_from_fork  : False
Retrieved bucket info:
  external_repo: 
  bucket       : therock-ci-artifacts
.Manifest expected at: /tmp/tmpw1ful3he/manifests/therock-manifest_jax_py3.12_nightly.json
.Manifest expected at: /tmp/tmppz6b5gnb/manifests/therock-manifest_jax_py3.12_release-0.4.28.json
Retrieving bucket info...
  (implicit) github_repository: ROCm/TheRock
  (implicit) is_pr_from_fork  : False
Retrieved bucket info:
  external_repo: 
  bucket       : therock-ci-artifacts
.Manifest expected at: /tmp/tmpham38j4t/manifests/therock-manifest_jax_py3.12_nightly.json
Retrieving bucket info...
  (implicit) github_repository: ROCm/TheRock
  (implicit) is_pr_from_fork  : False
Retrieved bucket info:
  external_repo: 
  bucket       : therock-ci-artifacts
.......
----------------------------------------------------------------------
Ran 10 tests in 0.004s

OK
```

```
TheRock$ python -m unittest build_tools/github_actions/tests/upload_jax_manifest_test.py
Manifest expected at: /tmp/tmp_up_r3p9/manifests/therock-manifest_jax_py3.12_nightly.json
Retrieving bucket info...
  (implicit) github_repository: ROCm/TheRock
  (implicit) is_pr_from_fork  : False
Retrieved bucket info:
  external_repo: 
  bucket       : therock-ci-artifacts
.Manifest expected at: /tmp/tmpzgbga5ay/manifests/therock-manifest_jax_py3.12_nightly.json
.Manifest expected at: /tmp/tmpwv25nfud/manifests/therock-manifest_jax_py3.12_release-0.4.28.json
Retrieving bucket info...
  (implicit) github_repository: ROCm/TheRock
  (implicit) is_pr_from_fork  : False
Retrieved bucket info:
  external_repo: 
  bucket       : therock-ci-artifacts
.Manifest expected at: /tmp/tmphfdh3spk/manifests/therock-manifest_jax_py3.12_nightly.json
Retrieving bucket info...
  (implicit) github_repository: ROCm/TheRock
  (implicit) is_pr_from_fork  : False
Retrieved bucket info:
  external_repo: 
  bucket       : therock-ci-artifacts
.......
----------------------------------------------------------------------
Ran 10 tests in 0.004s
```
### CI
```
{
  "jax": {
    "commit": "164fc708cb2176b97a46e5bd6dd4fa804aeec2ac",
    "repo": "https://github.com/ROCm/jax",
    "branch": "rocm-jaxlib-v0.8.2"
  },
  "therock": {
    "commit": "af6c5f749e6685af2bbd91f1f85404797b496840",
    "repo": "https://github.com/ROCm/TheRock.git",
    "branch": "users/erman-gurses/jax-manifest-followup"
  }
}
```
https://therock-dev-artifacts.s3.us-east-2.amazonaws.com/22987110999-linux/manifests/gfx94X-dcgpu/therock-manifest_jax_py3.12_rocm-jaxlib-v0.8.2.json

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

**Comment by kiran-thumma** [comment]
TODO: Enhancement can be done later good to have a single manifest generation file for both pytorch and jax. So we dont duplicate code.

**Comment by erman-gurses** [comment]
> TODO: Enhancement can be done later good to have a single manifest generation file for both pytorch and jax. So we dont duplicate code.

That is the plan actually - I will add more common functions step by step from the all manifest generations (TheRock, Pytorch, Jax) into the util file.
